api/v1beta1/argocd_types.go

```diff
@@ -1164,20 +1164,158 @@ type PrincipalSpec struct {
 	// Enabled is the flag to enable the Principal component during Argo CD installation. (optional, default `false`)
 	Enabled *bool `json:"enabled,omitempty"`
 
-	// AllowedNamespaces is the list of namespaces that the Principal component is allowed to access.
-	AllowedNamespaces []string `json:"allowedNamespaces,omitempty"`
+	// Server defines the server options for the Principal component.
+	Server *PrincipalServerSpec `json:"server,omitempty"`
+
+	// Namespace is the configuration for the Principal component namespace.
+	Namespace *PrincipalNamespaceSpec `json:"namespace,omitempty"`
+
+	// ResourceProxy defines the Resource Proxy options for the Principal component.
+	ResourceProxy *PrincipalResourceProxySpec `json:"resourceProxy,omitempty"`
+
+	// Redis defines the Redis options for the Principal component.
+	Redis *PrincipalRedisSpec `json:"redis,omitempty"`
 
-	// JWTAllowGenerate is the flag to enable the JWT generation during Argo CD installation.
-	JWTAllowGenerate bool `json:"jwtAllowGenerate,omitempty"`
+	// TLS defines the TLS options for the Principal component.
+	TLS *PrincipalTLSSpec `json:"tls,omitempty"`
+
+	// JWT defines the JWT options for the Principal component.
+	JWT *PrincipalJWTSpec `json:"jwt,omitempty"`
+}
+
+type PrincipalServerSpec struct {
 
 	// Auth is the authentication method for the Principal component.
 	Auth string `json:"auth,omitempty"`
 
-	// LogLevel refers to the log level used by the Principal component. Defaults to info if not configured. Valid options are debug, info, trace, error, and warn.
+	// MetricsPort is the port number for the Principal component metrics.
+	MetricsPort int `json:"metricsPort,omitempty"`
+
+	// PprofPort is the port number for the Principal component pprof.
+	PprofPort int `json:"pprofPort,omitempty"`
+
+	// HealthzPort is the port number for the Principal component healthz.
+	HealthzPort int `json:"healthzPort,omitempty"`
+
+	// ListenHost is the interface address to listen on.
+	ListenHost string `json:"listenHost,omitempty"`
+
+	// ListenPort is the port gRPC server should listen on.
+	ListenPort int `json:"listenPort,omitempty"`
```

Comment:
Not sure these fields are worth customizing at this time. If we do, we need to update Service.